### ISSUE TITLE
Travis CI builds failing on ppc64le - error building numpy

### ISSUE THREAD

**Issue by takluyver** [h5py/h5py#2030]
It appears this has been occurring since NumPy 1.21.5 was released in December. The [last successful build](https://app.travis-ci.com/github/h5py/h5py/jobs/552832738) on December 19th picked up NumPy 1.21.4.

```
    File "setup.py", line 448, in <module>
      setup_package()
    File "setup.py", line 440, in setup_package
      setup(**metadata)
    File "/tmp/pip-install-xetrj4b8/numpy_6b34dfa3f3cd4900a1b8b9fb53e94293/numpy/distutils/core.py", line 169, in setup
      return old_setup(**new_attr)
    File "/tmp/pip-build-env-knvp9iv8/overlay/lib/python3.7/site-packages/setuptools/__init__.py", line 153, in setup
      return distutils.core.setup(**attrs)
    File "/opt/python/3.7.1/lib/python3.7/distutils/core.py", line 148, in setup
      dist.run_commands()
    File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 966, in run_commands
      self.run_command(cmd)
    File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/tmp/pip-build-env-knvp9iv8/overlay/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 299, in run
      self.run_command('build')
    File "/opt/python/3.7.1/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmdobj.run()
    File "/tmp/pip-install-xetrj4b8/numpy_6b34dfa3f3cd4900a1b8b9fb53e94293/numpy/distutils/command/build.py", line 61, in run
      old_build.run(self)
    File "/opt/python/3.7.1/lib/python3.7/distutils/command/build.py", line 135, in run
      self.run_command(cmd_name)
    File "/opt/python/3.7.1/lib/python3.7/distutils/cmd.py", line 313, in run_command
      self.distribution.run_command(command)
    File "/opt/python/3.7.1/lib/python3.7/distutils/dist.py", line 985, in run_command
      cmd_obj.run()
    File "/tmp/pip-install-xetrj4b8/numpy_6b34dfa3f3cd4900a1b8b9fb53e94293/numpy/distutils/command/build_src.py", line 144, in run
      self.build_sources()
    File "/tmp/pip-install-xetrj4b8/numpy_6b34dfa3f3cd4900a1b8b9fb53e94293/numpy/distutils/command/build_src.py", line 155, in build_sources
      self.build_library_sources(*libname_info)
    File "/tmp/pip-install-xetrj4b8/numpy_6b34dfa3f3cd4900a1b8b9fb53e94293/numpy/distutils/command/build_src.py", line 288, in build_library_sources
      sources = self.generate_sources(sources, (lib_name, build_info))
    File "/tmp/pip-install-xetrj4b8/numpy_6b34dfa3f3cd4900a1b8b9fb53e94293/numpy/distutils/command/build_src.py", line 378, in generate_sources
      source = func(extension, build_dir)
    File "numpy/core/setup.py", line 684, in get_mathlib_info
      f"Broken toolchain: cannot link a simple {lang.upper()} "
  RuntimeError: Broken toolchain: cannot link a simple C++ program. note: A compiler with support for C++11 language features is required.
```

**Comment by takluyver** [comment]
The error message comes from a new check added in NumPy 1.21.5. It seems to be too strict, and fail where building NumPy would otherwise work, so I've filed a NumPy issue at https://github.com/numpy/numpy/issues/20761 .

**Comment by takluyver** [comment]
Fixed in #2031.
